pkg/pillar/hardware/smart.go

```diff
@@ -169,20 +225,38 @@ func GetInfoFromNVMeDisk(diskName string) (*types.DiskSmartInfo, error) {
 		return diskInfo, diskInfo.Errors
 	}
 
-	smartTemperature := new(types.DAttrTable)
-	smartTemperature.ID = types.SmartAttrIDTemperatureCelsius
-	smartTemperature.RawValue = int(smartAttr.Temperature)
-	diskInfo.SmartAttrs = append(diskInfo.SmartAttrs, smartTemperature)
+	// processSmartAttributes extracts SMART attributes from the NvmeSMARTLog struct and stores them in diskInfo.
+	// Use reflection to inspect the structure of smartAttr at runtime
+	val := reflect.ValueOf(*smartAttr) // Get the actual values of the fields
```

Comment:
Why do you need to inspect the type? Does dev.ReadSMART() return an interface? The old code treats it as a struct with fields like Temperature.

Reply:
OK, as long as the total number of counters will not make the message size explode for some storage devices. If so we would need some logic to figure out which subset of variables to return.

Reply:
I see on my laptop I have an NVME which reports Error Information which might be an array/slice of strings (can't tell since it doesn't have any errors at the moment). Wouldn't users be interested in such info?

Maybe in a follow-on PR we should add some specific handling of arrays/slices which might be important (in addition to the current default handling of different integer types.